### DISCUSSION TITLE
Relative to overwrite path params

### DISCUSSION THREAD

**Discussion by shubhambajad** [apideck-libraries/portman#380] [Q&A] [answered]
Hi Team,

We have one api request in which path parameter is mandatory for ex.
www.test.com/test/test_id/fields?page=1
and for such type of request I want to validate negative scenario by not passing the path params.

So, I have used variation testing in which I tried to overwrite the path params by passing below codes but it is not working

1. it is not passing a blank value if we have only passed "".
"overwriteRequestPathVariables": [
                  {
                    "key": "test_id",
                    "value": "",
                    "overwrite": true
                }

2. it is not passing working if we have only passed null.
"overwriteRequestPathVariables": [
                  {
                    "key": "test_id",
                    "value": null,
                    "overwrite": true
                }

3. it is working when we pass " " but in that case server consider it as a space and gives us the result 404 
but we want 400 as we are not passing the value.

So is there any way to pass the value as blank not even space?

**Comment by thim81** [comment, marked as answer]
@shubhambajad This seems like a bug.

I have created an issue for this and will see how to fix this.

**Reply by shubhambajad** [reply]
@thim81 Also can you check whether the pathparams "remove":true property is working or not? because I have tried it but it was not working, variable was remain as it.

**Reply by thim81** [reply]
The fix that to support "" has landed, and will be part of the next release.

I'll look into the "remove:true" property.

**Reply by thim81** [reply]
@shubhambajad You were correct, the remove option was missing from the implementation.

**Reply by shubhambajad** [reply]
@thim81 Thanks for confirming ..

**Reply by thim81** [reply]
@shubhambajad We just pushed a PR that will bring overwriteRequestPathVariables - insert/remove path variables

**Reply by thim81** [reply]
@shubhambajad In the latest Portman release (version 1.18.0) this feature is added, including the support of empty values.

**Reply by shubhambajad** [reply]
@thim81 Thank you !!
